### PR TITLE
Preprocessor to sign (trust) notebooks

### PR DESCRIPTION
**Pull request by torosdahl**
Preprocessor to sign notebooks. Replaces the jupyter trust command in the git smudge filter.

### CODE REVIEW

**Comment by akhmerov** [comment]
All looks great. Can you also test it with master nbconvert and using `--stdin` flag. Then both filters would boil down to a single command, and we wouldn't need temporary files anymore.

Another thing, I'd call `SignNBPreprocessor` `TrustPreprocessor`. Also I'm not sure if you need to check signature before signing: I think this does the work of calculating the hash twice.

**Comment by torosdahl** [comment]
stdin works with nbconvert master!

File: scripts/nb_filter_clean.sh [hunk 1]
```diff
@@ -1,5 +1,2 @@
 #!/bin/bash
-temp_out=$(mktemp)
-cat > $temp_out
-jupyter nbconvert --to notebook --config scripts/config_filter_clean.py $temp_out --stdout
-rm $temp_out
+jupyter nbconvert --to notebook --config scripts/config_filter_clean.py $temp_out --stdin --stdout
```
Comment (akhmerov) [inline scripts/nb_filter_clean.sh]: Does this work? `$temp_out` doesn't belong here.

**Comment by torosdahl** [comment]
Whoops... taken care of now.

**Pull request merged**